test/env/setup_env.sh

```diff
@@ -106,16 +106,11 @@ case "$SUNDIALS_TEST_TYPE" in
         ;;
 esac
 
-# C and C++ standards. Note only double precision math functions are supported
-# in ISO C89/C90. While profiling requires C99 for __func__ some compilers still
-# define __func__ with ISO C89/C90. As such we do not consider if profiling is
-# enabled and let the environment script override this setting if necessary.
-if [ "${SUNDIALS_TPLS}" == "OFF" ] && [ "${SUNDIALS_PRECISION}" == "double" ]
-then
-    export CMAKE_C_STANDARD="90"
-else
-    export CMAKE_C_STANDARD="99"
-fi
+# Build Type
+export CMAKE_BUILD_TYPE="Debug"
+
+# C and C++ standards.
+export CMAKE_C_STANDARD="99"
```

Comment:
It looks like this PR requires building a subset of C99 features, is that right?

Reply:
Indeed. It uses `__func__`, `__FILE__` and `__LINE__` for one.

Reply:
Also, it requires building sundials with a subset of C99, but should not require user code to be compiled with C99.

Reply:
Right, we'll want to add a CI test that builds the examples with TPLs off and C90 just in case a `//` comment or something else sneaks in to a header accidentally. We could update the `test/test_install.py` script for this. It would need a minor update to set the C standard (overriding what's in the example's `CMakeLists.txt`) but otherwise should do what we need.